src/app/core/learning-object-module/bundling/bundling.service.ts

```diff
@@ -39,4 +50,49 @@ export class BundlingService {
       return throwError(error);
     }
   }
+
+  toggleBundle(
+    learningObjectId: string,
+    fileIDs: string[],
+    state: boolean
+  ) {
+    const route = BUNDLING_ROUTES.TOGGLE_BUNDLE_FILE({ learningObjectId });
+
+    return this.http
+      .patch(
+        route,
+        {
+          fileIDs: fileIDs,
+          packagable: state
+        },
+        { headers: this.headers, withCredentials: true }
+      )
+      .pipe(
+
+        catchError(this.handleError)
+      )
+      .toPromise();
+  }
+
+
+  toggleDownloadModal(val?: boolean) {
+    this.showDownloadModal = val;
+  }
+
+  download(learningObjectId: string) {
+    this.toggleDownloadModal(true);
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
+  }
+
+
+  downloadObject(event: MouseEvent, object: LearningObject, index: number) {
+    event.stopPropagation();
+    this.currentIndex = index;
+    this.downloading[index] = true;
+    this.showDownloadModal = true;
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(object.id));
```

Comment:
The downloadBundle function in libraryService should be moved to the bundling.service.ts file.

Reply:
+1